appeals/web/src/server/appeals/appeal-details/appellant-case/outcome-invalid/outcome-invalid.controller.js

```diff
@@ -0,0 +1,130 @@
+import logger from '../../../../lib/logger.js';
+import * as appellantCaseService from '../appellant-case.service.js';
+import {
+	mapInvalidOrIncompleteReasonsToCheckboxItemParameters,
+	mapPostedReviewOutcomeToApiReviewOutcome
+} from '../appellant-case.mapper.js';
+import { objectContainsAllKeys } from '../../../../lib/object-utilities.js';
+
+/**
+ *
+ * @param {import('@pins/express/types/express.js').Request} request
+ * @param {import('@pins/express/types/express.js').RenderedResponse<any, any, Number>} response
+ */
+const renderInvalidReason = async (request, response) => {
+	const { errors, body } = request;
+
+	if (!objectContainsAllKeys(request.session, ['appealId', 'appealReference'])) {
+		return response.render('app/500.njk');
+	}
+
+	const {
+		// @ts-ignore
+		appealId,
+		// @ts-ignore
+		appealReference,
+		// @ts-ignore
+		appellantCaseReviewOutcome
+	} = request.session;
+	const invalidReasonOptions = await appellantCaseService.getAppellantCaseInvalidReasons();
+
+	if (invalidReasonOptions) {
+		const invalidReasons = body.invalidReason || appellantCaseReviewOutcome?.invalidReasons;
+		const otherReason = body.otherReason || appellantCaseReviewOutcome?.otherNotValidReasons;
+		const mappedInvalidReasonOptions = mapInvalidOrIncompleteReasonsToCheckboxItemParameters(
+			invalidReasonOptions,
+			invalidReasons
+		);
+
+		const appealReferenceFragments = appealReference.split('/');
+
+		return response.render('appeals/appeal/appellant-case-invalid-incomplete.njk', {
+			appeal: {
+				id: appealId,
+				shortReference: appealReferenceFragments?.[appealReferenceFragments.length - 1]
+			},
+			notValidStatus: 'invalid',
+			reasonOptions: mappedInvalidReasonOptions,
+			otherReasonId: invalidReasonOptions.find((reason) => reason.name.toLowerCase() === 'other')
```

Comment:
This "find 'other'" logic is repeated, do we need a little util function?

Reply:
I've moved this into a util function